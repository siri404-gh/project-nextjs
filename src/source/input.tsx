import React, { useState } from 'react';

export const Input = () => {

  const [ val, setVal ] = useState('');

  const onInputChange = e => {
    e.preventDefault();
    setVal(e.target.value);
  };

  return <input type='text' onChange={onInputChange} value={val} />
};
