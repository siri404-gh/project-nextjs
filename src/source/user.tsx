import React, { useState, useEffect } from 'react';

// const fakeUser = {
//   name: "Sreeram Padmanabhan",
//   age: "32",
//   address: "123, Charming Avenue"
// };

// const fetch = url => Promise.resolve({
//   json: () => Promise.resolve(fakeUser),
// });

// const fetch = url => new Promise((resolve, reject) => {
//   resolve({
//     json: () => new Promise((resolve, reject) => {
//       resolve(fakeUser);
//     })
//   })
// });

export default function User() {
  const [ user, setUser ] = useState(null);

  useEffect(() => {
    (async function fetchUserData() {
      const response = await fetch('https://jsonplaceholder.typicode.com/todos/1');
      setUser(await response.json());
    })();
  }, []);

  if (!user) {
    return "loading...";
  }

  return (
    <pre>
      {JSON.stringify(user)}
    </pre>
  );
}
