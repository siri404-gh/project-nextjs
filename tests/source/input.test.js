import { render, unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { Input } from '@/source/input';

describe('Index with act', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Input />, container);
    });
    expect(container).toMatchSnapshot();
  });

});

describe('Input with enzyme', () => {
  it('Should set value to state when input is changed', () => {
    const container = shallow(<Input />);
    const input = container.find('input');
    input.simulate('change', { preventDefault: jest.fn, target: { value: "foo" } });
    expect(container).toMatchSnapshot();
  });
});
