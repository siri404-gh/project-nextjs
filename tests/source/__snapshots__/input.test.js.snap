// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index with act Should render without props 1`] = `
<section>
  <input
    type="text"
    value=""
  />
</section>
`;

exports[`Input with enzyme Should set value to state when input is changed 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="foo"
/>
`;
